src/ophyd_async/core/_signal.py

```diff
@@ -135,7 +138,9 @@ def _notify(
         function: Callback[dict[str, Reading[SignalDatatypeT]] | SignalDatatypeT],
         want_value: bool,
     ):
-        assert self._reading, "Monitor not working"
+        if not self._reading:
+            msg = "Monitor not working"
+            raise RuntimeError(msg)
```

Comment:
Same as above, is it worth making a utility function for this?

Reply:
given that we can't re-use the `get_reading` because it's async, but the `_callback` and `_notify` are both synchronous, I don't really the the usefulness of a dedicated utility method. Unless I misunderstood you.

Reply:
I wondered if it was worth doing this:
```python
def _ensure_reading(self):
        if not self._reading:
            msg = "Monitor not working"
            raise RuntimeError(msg)
        return self._reading
```
Then in the 2 places we use `self._reading` we instead do `reading = self._ensure_reading()`

Reply:
Sure, that makes it much nicer to read!
added here 5098783e629a91675d6ee133d831f4e2493ad6a3
test have been added for both happy cases, the RuntimeError was already there.

also added missing type hints for the class
a769e576f40286be50acd786e1ce285b96d658d8